File: 2023/advent_of_code/test/q23/q23_test.exs
defmodule AdventOfCode.Q23Test do
  use ExUnit.Case
  doctest AdventOfCode.Q23

  test "Q23 Part 1 Example works" do
    file_stream = File.stream!("./test/q23/q23_example.txt")
    result = AdventOfCode.Q23.part1(file_stream)
    assert result == 94
  end

  test "Q23 Part 1 works" do
    file_stream = File.stream!("./test/q23/q23.txt")
    result = AdventOfCode.Q23.part1(file_stream)
    assert result == 1998
  end

  test "Q23 Part 2 Example works" do
    file_stream = File.stream!("./test/q23/q23_example.txt")
    result = AdventOfCode.Q23.part2(file_stream)
    assert result == 154
  end

  test "Q23 Part 2 works" do
    file_stream = File.stream!("./test/q23/q23.txt")
    result = AdventOfCode.Q23.part2(file_stream)
    assert result == 6434
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q23/Q23Test.kt
package q23

import junit.framework.TestCase.assertEquals
import java.nio.file.Paths
import kotlin.test.Test

class Q23Test {
    @Test
    fun `part 1 works`()  {
        val filePath = javaClass.getResource("/q23.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(4249 , part1(filePath!!) )
    }

    @Test
    fun `part 2 works`()  {
        val filePath = javaClass.getResource("/q23.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(980, part2(filePath!!) )
    }
}